fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java

```diff
@@ -1829,9 +1829,17 @@ public PlanFragment visitPhysicalNestedLoopJoin(
     public PlanFragment visitPhysicalLimit(PhysicalLimit<? extends Plan> physicalLimit, PlanTranslatorContext context) {
         PlanFragment inputFragment = physicalLimit.child(0).accept(this, context);
         PlanNode child = inputFragment.getPlanRoot();
-        child.setLimit(MergeLimits.mergeLimit(physicalLimit.getLimit(), physicalLimit.getOffset(), child.getLimit()));
-        // TODO: plan node don't support limit
-        // child.setOffset(MergeLimits.mergeOffset(physicalLimit.getOffset(), child.getOffset()));
+
+        if (!(child instanceof ExchangeNode)) {
```

Comment:
1. remove post process: AddOffsetIntoDistribute
2. in translator
2.1. if limit is local, just merge it with child.getLimit() and set it to child.limit
2.2. if limit is global, check wether it's child is exchange. if child is exchange, just merge limit and offset with exchange's limit and offset. if child is not exchange, create exchange node and set limit and offset to it

Reply:
done